### PR TITLE
MAT-9632 validate harp Id when sharing unsharing

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-9632](https://jira.cms.gov/browse/MAT-9632)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Comment by sb-cecilialiu** [comment]
MAT-9632_validateUser_withUserHarpId

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx [hunk 1]
```diff
@@ -179,6 +179,22 @@ const LibraryShareDialog = ({
...
+    try {
+      const userDetails = await userServiceApi.getOwnerDetails(
+        harpId.toLowerCase()
+      );
+      if (userDetails && UserStatus[0] !== userDetails.userStatus.toString()) {
+        throw new Error("User is not active");
+      }
+    } catch (error) {
+      // set error for harpId field
+      formik.setFieldError(
```
Comment (adongare) [inline src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: I think we need to raise the field error based on http status code. @etan-sb did something similar for measure side- https://github.com/MeasureAuthoringTool/madie-measure/blob/90ed5291312fc12c4d61e6c73b7096dd3878166e/src/components/common/shareDialog/ShareDialog.tsx#L207

Comment (etan-sb) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: Here are the PRs I worked for [MAT-8218](https://jira.cms.gov/browse/MAT-8218) (Share Measure - User Validation - Admin API & UI (front end & back end)):
https://github.com/MeasureAuthoringTool/madie-measure/pull/1483
https://github.com/MeasureAuthoringTool/madie-util/pull/188
https://github.com/MeasureAuthoringTool/measure-service/pull/1094

Ashok suggested a comment where we want to display a generic toast when a non-400 code returns. If you also want to implement this in your PR, you can just follow the similar implementation in madie-measure and madie-util (needed change to rethrow the Axios error with the full response rather than swallowing the error and throwing a generic error message).

Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: @adongare @etan-sb I added handling for other statuses. Thanks a lot for the recommendation.

Comment (adongare) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: thank you, @sb-cecilialiu! I dont see the change. by any chance forgot to push the change?

Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: @adongare I thought I did, looks like I didn't. I just pushed it. Thanks!

**Review by adongare — APPROVED** [review]
